Core/automation/lib/python/core/utils.py

```diff
@@ -118,6 +122,61 @@ def validate_uid(uid):
     uid = re.sub(r"__+", "_", uid)
     return uid
 
+def get_item_value(item_or_item_name, return_type=None, default=None, unit=None):
+    """
+    Returns the Item's value if the Item exists and is initialized, otherwise default. 
+    Returned item type will be dependent on the item's type. If return_type is supplied,
+    the Item will be returned as that type if possible, otherwise the default will be 
+    returned. Item value will be converted to the provided unit type if possible; otherwise,
+    the default will be returned. If no default is supplied and an Item value cannot be 
+    returned, None is returned 
+
+    Args:
+        item_or_item_name (Item or str): name of the Item
+        return_type (int, float, str): item type to be returned. If the Item cannot be 
+            converted to return_type, default will be returned 
+        default (int, float, ON, OFF, OPEN, CLOSED, str, DateTime): the default
+            value to be returned if the Item is non-existent/uninitialized or cannot
+            be converted to return_type
+        unit (QuantityType): Convert value of Item to specified units
+            https://www.openhab.org/docs/concepts/units-of-measurement.html
+
+    Returns:
+        int, float, ON, OFF, OPEN, CLOSED, str, DateTime, or None: the state of the Item
+            as a suitable type, or converted to return_type if given; or the default value 
+            if the Item's state cannot be converted or is NULL or UNDEF; otherwise None
+    """
```

Comment:
How about this:

```python
    """
    Get's an Item's state, optionally converting it to a specified type and/or unit for Quantity Types.

    Args:
        item_or_item_name (GenericItem or str): Item or name of the Item
        return_type (Any): Python or Java data type to attempt to return the Item's state as, if not
            provided or `None` no conversion will be done. If the conversion fails the default will
            be returned.
        default (Any): value to return if Item does not exist, if Item's state is ``NULL`` or ``UNDEF``
            or if attempts to convert Item's state fail.
        unit (str): if Item's state is a QuantityType, attempt to convert to this unit. This is done
            before any conversion attempts if ``return_type`` is provided.

    Returns:
        Item's state if Item exists and it's state is not ``NULL`` or ``UNDEF`` if no ``return_type``
        or ``unit`` is provided, otherwise ``default`` will be returned.
        If ``return_type`` is provided the Item's state will be returned as that type if possible,
        otherwise ``default`` will be returned.
        If ``unit`` is provided and the Item's state is a QuantityType it will be converted to those
        units before attempting to convert to ``return_type``, if the unit conversion fails ``defualt``
        will be returned.
    """
```

You will need to adjust the above for line length, there is no ruler in this comment box.
Don't forget to switch the order of the arguments if you change them in the function signature.

Reply:
I really like the idea of the URL to the explanatory doco for QuantityType, because It's quite difficult to find the relevant information.

Reply:
Much clearer! I think I'd forgotten the proper docstring format.

Reply:
I meant to put the link in actually, I agree it's a good reference to have!

> Much clearer! I think I'd forgotten the proper docstring format.

Practice makes perfect as they say, I wrote all the docs for MQTTany, Eos, and Selector among others.